### PR TITLE
Add custom certificate documentation

### LINKED ISSUE #398: Document the use of custom certificates for requests
As initially raised in https://github.com/stac-utils/pystac-client/issues/383, the incantation to provide a custom certificate to API requests is a little complicated:

```python
from pystac_client.stac_api_io import StacApiIO
from pystac_client.client import Client
stac_api_io = StacApiIO()
stac_api_io.session.verify = "/path/to/certfile"
client = Client.from_file("https://planetarycomputer.microsoft.com/api/stac/v1", stac_io=stac_api_io)
```

It'd be nice to include this example in the documentation.

### PR DESCRIPTION
**Pull request by gadomski**
**Related Issue(s):** 

- Closes #398 

**Description:**

Initially raised in #383, and worth documenting IMO.

**PR Checklist:**

- [x] Code is formatted
- [x] Tests pass
- [ ] Changes are added to the [CHANGELOG](https://github.com/stac-utils/pystac-api-client/blob/main/CHANGELOG.md)

### CODE REVIEW

File: docs/usage.rst [hunk 1]
```diff
@@ -323,6 +323,21 @@ object in-place and returns no result. A warning is emitted if your
...
+customize the :class:`StacApiIO<pystac_client.stac_api_io.StacApiIO>` instance before
+creating your :class:`Client<pystac_client.Client>`.
+
+.. code-block:: python
+
+    >>> from pystac_client.stac_api_io import StacApiIO
+    >>> from pystac_client.client import Client
+    >>> stac_api_io = StacApiIO()
+    >>> stac_api_io.session.verify = "/path/to/certfile"
+    >>> client = Client.from_file("https://planetarycomputer.microsoft.com/api/stac/v1", stac_io=stac_api_io)
```
Comment (pjhartzell) [inline docs/usage.rst:339]: Is there a reason why we are using `Client.from_file()` rather than `Client.open()`?

**Review by pjhartzell — CHANGES_REQUESTED** [review]

File: docs/usage.rst (hunk 1 above)
Comment (gadomski) [reply docs/usage.rst:339]: `open` doesn't take a `stac_io`: https://pystac-client.readthedocs.io/en/stable/api.html#client. Which seems like an inconsistency to me -- maybe it'd be worth adding `stac_io` to `open`? What do you think?

Comment (pjhartzell) [reply docs/usage.rst:339]: It looks like it does:
https://github.com/stac-utils/pystac-client/blob/7d589e802eba2499cfd97c2948a2ed79856937fd/pystac_client/client.py#L90-L101

Comment (gadomski) [reply docs/usage.rst:339]: Yup, looks like it's unreleased, was added here: https://github.com/stac-utils/pystac-client/pull/372. I'll update this example to use `open`.

**Pull request merged**